processout-sdk/src/main/java/com/processout/processout_sdk/ProcessOut.java

```diff
@@ -232,18 +233,53 @@ public void error() {
                     }
                 });
                 break;
+            case URL:
+                WebView theWebPage = new WebView(this.context);
+                theWebPage.getSettings().setJavaScriptEnabled(true);
+                theWebPage.getSettings().setPluginState(WebSettings.PluginState.ON);
+                Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(cA.getValue()));
+                this.context.startActivity(browserIntent);
+                break;
             default:
-                //TODO: handle FINGERPRINT, URL and REDIRECT for mobiles that don't support 3ds2 challenges
-                handler.onError(null);
+                handler.onError(new ProcessOutAuthException("Unhandled three D S action:" + cA.getType().name()));
                 break;
         }
     }
 
 
+    /**
+     * Checks if the opening URL is from ProcessOut and returns the corresponding value if so. Returns null otherwise
+     *
+     * @param intentData intentData from the Activity
+     * @return A WebViewReturnAction containing the return type, value and success or null if not from ProcessOut
+     */
+    public static WebViewReturnAction handleProcessOutReturn(Uri intentData) {
+        if (!intentData.getHost().contains("processout"))
```

Comment:
Are we sure the host will contain processout?

Reply:
Just checked this. The host here is the return URL of the invoice. As it's specified in the documentation it should be something like: `youapp://processout.return` with the host here being `processout.return`